ampd/src/queue/queued_broadcaster.rs

```diff
@@ -214,7 +213,7 @@ mod test {
         assert!(handle.await.unwrap().is_ok());
     }
 
-    #[test]
+    #[test(start_paused = true)]
     async fn should_not_broadcast_when_gas_limit_has_not_been_reached() {
```

Comment:
this test doesn't actually test what it should:
- the MockBroadcaster broadcast function should not be called at all
- we should advance the interval after the messages are queued for broadcast to make sure the ticker doesn't trigger a broadcast
- no drop(client)
- no waiting for the handle to complete at the end
- do a short sleep (50ms or something like that) to make sure the broadcaster had every chance to be triggered before the test ends

Reply:
Created a ticket for this to come back to later.